### PR TITLE
AzCli support for checkpoint operations and added deleteFromHost flag for VM delete

### PR DESCRIPTION
**Pull request by mchoubey2021**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az scvmm vm create-checkpoint --subscription contoso-sub \
--resource-group contoso-rg --name contoso-vm --checkpoint-name contoso-chkpt-name \
--checkpoint-description contoso-chkpt-description

 az scvmm vm delete-checkpoint --subscription contoso-sub \
--resource-group contoso-rg --name contoso-vm --checkpoint-id checkpoint-guid

 az scvmm vm restore-checkpoint --subscription contoso-sub \
--resource-group contoso-rg --name contoso-vm --checkpoint-id checkpoint-guid

az scvmm vm delete --subscription contoso-sub \
--resource-group contoso-rg --name contoso-avset --deleteFromHost

### General Guidelines

- [ ] Have you run `azdev style <scvmm>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/scvmm/azext_scvmm/_help.py [hunk 1]
```diff
@@ -684,6 +689,43 @@
...
+               az scvmm vm create-checkpoint --subscription contoso-sub \
+--resource-group contoso-rg --name contoso-vm --checkpoint-name contoso-chkpt-name \
+--checkpoint-description contoso-chkpt-description
+"""
+
+helps[
+    'scvmm vm delete-checkpoint'
+] = """
+    type: command
+    short-summary: Deletes the specified VM checkpoint
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_help.py:709]:
```suggestion
    short-summary: Delete the specified VM checkpoint
```

File: src/scvmm/azext_scvmm/_help.py [hunk 2]
```diff
@@ -684,6 +689,43 @@
...
+--resource-group contoso-rg --name contoso-vm --checkpoint-id checkpoint-guid
+"""
+
+helps[
+    'scvmm vm restore-checkpoint'
+] = """
+    type: command
+    short-summary: Restores to VM checkpoint
+    examples:
+      - name: Restores VM checkpoint
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_help.py:723]:
```suggestion
    short-summary: Restore to VM checkpoint
    examples:
      - name: Restore VM checkpoint
```

Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_help.py:723]:
```suggestion
    short-summary: Restore VM checkpoint
    examples:
      - name: Restore VM checkpoint
```

File: src/scvmm/azext_scvmm/_params.py [hunk 3]
```diff
@@ -170,6 +170,33 @@ def load_arguments(self: AzCommandsLoader, _):
             help="Skip shutdown and power-off immediately.",
         )
 
+    with self.argument_context('scvmm vm create-checkpoint') as c:
+        c.argument(
+            'checkpoint_name',
+            options_list=['--checkpoint-name'],
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_params.py:176]:
```suggestion
```

File: src/scvmm/azext_scvmm/_params.py [hunk 4]
```diff
@@ -170,6 +170,33 @@ def load_arguments(self: AzCommandsLoader, _):
...
+            help="Name of the checkpoint to be created.",
+        )
+
+        c.argument(
+            'checkpoint_description',
+            options_list=['--checkpoint-description'],
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_params.py:182]:
```suggestion
```

File: src/scvmm/azext_scvmm/_params.py [hunk 5]
```diff
@@ -170,6 +170,33 @@ def load_arguments(self: AzCommandsLoader, _):
...
+            help="Description of the checkpoint to be created.",
+        )
+
+    with self.argument_context('scvmm vm delete-checkpoint') as c:
+        c.argument(
+            'checkpoint_id',
+            options_list=['--checkpoint-id'],
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_params.py:189]:
```suggestion
```

File: src/scvmm/azext_scvmm/_params.py [hunk 6]
```diff
@@ -170,6 +170,33 @@ def load_arguments(self: AzCommandsLoader, _):
...
+            help="Checkpoint UUID.",
+        )
+
+    with self.argument_context('scvmm vm restore-checkpoint') as c:
+        c.argument(
+            'checkpoint_id',
+            options_list=['--checkpoint-id'],
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/_params.py:196]:
```suggestion
```

**Comment by yanzhudd** [comment]
Please add test for the new added commands

**Comment by mchoubey2021** [comment]
Added test cases for newly added scenarios

File: src/scvmm/azext_scvmm/tests/latest/test_scvmm_scenario.py [hunk 7]
```diff
@@ -151,7 +153,22 @@ def test_scvmm(self):
...
+
+        self.cmd(
+            'az scvmm vm restore-checkpoint -g {resource_group} --name {vm_name} --checkpoint-id {checkpoint_id}',
+        )
+
+        self.cmd(
+            'az scvmm vm delete-checkpoint -g {resource_group} --name {vm_name} --checkpoint-id {checkpoint_id}',
+        )
+
+        self.cmd('az scvmm vm delete -g {resource_group} --name {vm_name} --deleteFromHost -y')
```
Comment (yanzhudd) [inline src/scvmm/azext_scvmm/tests/latest/test_scvmm_scenario.py:171]: It is suggested to add some checks for the commands.

Comment (mchoubey2021) [reply src/scvmm/azext_scvmm/tests/latest/test_scvmm_scenario.py]: Added checks. Pulled vm states and made checks.

**Review by yanzhudd — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**